packages/builders/src/interactions/slashCommands/SlashCommandBuilder.ts

```diff
@@ -1,108 +1,40 @@
-import type {
-	APIApplicationCommandOption,
-	ApplicationIntegrationType,
-	InteractionContextType,
-	LocalizationMap,
-	Permissions,
-} from 'discord-api-types/v10';
-import { mix } from 'ts-mixer';
-import { SharedNameAndDescription } from './mixins/NameAndDescription.js';
+import type { JSONEncodable } from '@discordjs/util';
+import { ApplicationCommandType, type RESTPostAPIChatInputApplicationCommandsJSONBody } from 'discord-api-types/v10';
+import { Mixin } from 'ts-mixer';
+import { isValidationEnabled } from '../../util/validation.js';
+import { slashCommandPredicate } from './Assertions.js';
+import { SharedNameAndDescription } from './mixins/SharedNameAndDescription.js';
 import { SharedSlashCommand } from './mixins/SharedSlashCommand.js';
 import { SharedSlashCommandOptions } from './mixins/SharedSlashCommandOptions.js';
 import { SharedSlashCommandSubcommands } from './mixins/SharedSubcommands.js';
 
 /**
  * A builder that creates API-compatible JSON data for slash commands.
  */
-@mix(SharedSlashCommandOptions, SharedNameAndDescription, SharedSlashCommandSubcommands, SharedSlashCommand)
-export class SlashCommandBuilder {
+export class SlashCommandBuilder
+	extends Mixin(SharedSlashCommandOptions, SharedNameAndDescription, SharedSlashCommandSubcommands, SharedSlashCommand)
```

Comment:
Just curious, any reason for moving from decorator to function? Not that its an issue or a blocker, just curious

Reply:
Interface merging we did before didn't work with how I'm now handling the `data` field. There was probably a way, but I couldn't figure it out